src/main/java/org/mskcc/cmo/metadb/service/impl/UpdateMessageHandlingServiceImpl.java

```diff
@@ -0,0 +1,287 @@
+package org.mskcc.cmo.metadb.service.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.nats.client.Message;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Phaser;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.logging.log4j.util.Strings;
+import org.mskcc.cmo.messaging.Gateway;
+import org.mskcc.cmo.messaging.MessageConsumer;
+import org.mskcc.cmo.metadb.service.UpdateMessageHandlingService;
+import org.mskcc.cmo.metadb.service.ValidRequestChecker;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UpdateMessageHandlingServiceImpl implements UpdateMessageHandlingService {
```

Comment:
mentioned above already but I want to rename these - again this can be done outside of this pr and will not be part of the requested changes for this review